README.md

```diff
@@ -11,64 +11,51 @@ Uses request subdomain to swap database, Rails cache (preferences), image paths.
 
 ### Installation
 
-Add to your `Gemfile`:
+1. Add to your `Gemfile`:
 
-```ruby
-gem 'spree_shared', github: 'spree-contrib/spree_shared', branch: 'master'
-```
+    ```ruby
+    gem 'spree_shared', github: 'spree-contrib/spree_shared', branch: 'master'
+    ```
 
-Make sure your `config/database.yml` has valid db connection.
+    Make sure your `config/database.yml` has valid db connection.
 
-Create an initializer `config/initializers/apartment.rb` with the following command:
+2. Create `config/initializers/apartment.rb`with the following command:
 
-```bash
-bundle exec rails generate apartment:install
-```
+    ```bash
+    bundle exec rails generate apartment:install
+    ```
 
-Search for following line:
+3. Search for following line inside `config/initializers/apartment.rb`:
 
-```ruby
-Apartment.configure do |config|
-  ...
-  # supply list of database names for migrations to run on
-  config.tenant_names = lambda { ToDo_Tenant_Or_User_Model.pluck :database }
-end
-```
+    ```ruby
+      config.tenant_names = lambda { ToDo_Tenant_Or_User_Model.pluck :database }
+    ```
 
-And change it to include two sample subdomains:
+    And change it to include two sample subdomains:
 
-```ruby
-Apartment.configure do |config|
-  ...
-  config.tenant_names = %w(store1 store2)
-end
-```
+    ```ruby
+      config.tenant_names = %w(store1 store2)
+    ```
 
-Add the following line to host `application.rb`
+4. Bootstrap sample stores:
 
-```ruby
-config.middleware.use 'Apartment::Elevators::Subdomain'
-```
+    ```bash
+    bundle exec rake spree_shared:bootstrap['store1']
+    bundle exec rake spree_shared:bootstrap['store2']
+    ```
 
-Bootstrap sample stores:
+5. Setup local subdomains for sample stores, as spree_shared uses by default subdomain routing you need to confirm some local domains such as:
 
-```bash
-bundle exec rake spree_shared:bootstrap['store1']
-bundle exec rake spree_shared:bootstrap['store2']
-```
-
-Setup local subdomains for sample stores, as spree_shared uses by default subdomain routing you need to confirm some local domains such as:
-
-    store1.spree.dev
-    store2.spree.dev
+    `store1.spree.dev`
+    `store2.spree.dev`
+    
+    This can be done using [Pow][4] or editing your local `/etc/hosts` file.
 
-This can be done using [Pow][4] or editing your local `/etc/hosts` file.
+6. Set namespace for cache engine in `development.rb` and/or `production.rb`
```

Comment:
With this PR merged we probably should emphasise this is really obligatory to make things work, otherwise all the cache entries from different tenants would mix up together

Reply:
That's why I made an step by step installation list with "6. Set namespace for cache engine in `development.rb` and/or `production.rb`" as one of the obligatory points. Is that not enough you think?

Reply:
I am thinking about users that right now rely only on the `Spree::Preferences::ScopedStore#rails_cache-id` [source](https://github.com/spree-contrib/spree_shared/blob/master/app/models/spree_shared/scoped_store_decorator.rb#L5) with no cache namespace set at all. 

But, on the other hand, instruction is here for some time so we can assume they have already followed it

Reply:
On the other hand there may be users that rely on original Spree::Preferences::ScopedStore#rails_cache-id 
[source](https://github.com/spree/spree/blob/master/core/app/models/spree/preferences/scoped_store.rb#L30) and hope that `ENV['RAILS_CACHE_ID']` will be a part of their cache_key. They may not expect it missing from the key

Reply:
Explicitly asking to namespace their cache is much better than namespacing it silently and changing the wy cache_key is computed, throwing away `ENV['RAILS_CACHE_ID']`

Reply:
Keep in mind that the biggest problem this PR solves is that:

When setting preferences in initializer (like spree.rb) keys are different that the ones when getting preferences, because [spree_shared/scoped_store_decorator.rb](https://github.com/spree-contrib/spree_shared/blob/master/app/models/spree_shared/scoped_store_decorator.rb) is not yet loaded. In my case this caused getting default layout form spree instead fo the ones we set in initializer and 500